pravega/openmessaging-benchmark
# Issue 81: Customize pravega payload attributes before submitting to make each payload unique

**Change log description**
Added changes to customize pravega payload **Timestamp** attribute before submitting to Pravega, to make each payload unique.

**Purpose of the change**
Fixes https://github.com/pravega/openmessaging-benchmark/issues/81.

**What the code does**
This PR does the following:
Introduced new boolean configuration attribute **_customPayload_**.
If **customPayload** is set to true, then customize json request payload with Timestamp dynamic value in ms.

**How to verify it**
Enable  **customPayload** configuration for pravega and then run omb for pravega driver with sample payload **custom-payload-idrac.data** and workload **custom-payload-idrac.yaml** shared in this repository.
After that when you read those messages in pravega consumer check **Timestamp** attribute value, you will get different values almost each time in milliseconds. Or you can enble debug log in omb which will print customized payload for pravega, check **Timestamp** attribute value,it will print different values almost each time in milliseconds.
